sdk/core/Azure.Core.TestFramework/src/ClientTestFixtureAttribute.cs

```diff
@@ -34,41 +54,93 @@ public IEnumerable<TestSuite> BuildFrom(ITypeInfo typeInfo, IPreFilter filter)
         {
             if (_serviceVersions.Any())
             {
-                foreach (object serviceVersion in _serviceVersions)
+                if (_additionalParameters?.Any() ?? false)
                 {
-                    var syncFixture = new TestFixtureAttribute(false, serviceVersion);
-                    var asyncFixture = new TestFixtureAttribute(true, serviceVersion);
-
-                    foreach (TestSuite testSuite in asyncFixture.BuildFrom(typeInfo, filter))
+                    foreach (var parameter in _additionalParameters)
                     {
-                        Process(testSuite, serviceVersion, true);
-                        yield return testSuite;
+                        foreach (object serviceVersion in _serviceVersions)
+                        {
+                            var syncFixture = new TestFixtureAttribute(false, serviceVersion, parameter);
+                            var asyncFixture = new TestFixtureAttribute(true, serviceVersion, parameter);
```

Comment:
This looks like it'll only work for a single dimension of parameter.  If that's the case, maybe it'd be better to specialize this to `serviceEndpoints` vs. `additionalParameters`?

Reply:
Do we expect this to be used only for endpoints?

Reply:
I considered this, but there is nothing really specific to endpoints about the behavior. This is one reason I thought about implementing this as a different attribute based on the core functionality of "create test variations based on this value"..  I wonder if they would just be stackable in the case we wanted to use more than 1 variation dimension?

Reply:
They wouldn't stack without changing the code to wrap all of this in another loop per dimension.  Otherwise you'd just be testing every dimension value in isolation for each service version.  I almost wonder if this would make more sense to change to something like:

```C#
[ServiceConfiguration(ServiceVersion.Latest,   Endpoint = TablesEndpoints.Tables, Record = true)]
[ServiceConfiguration(ServiceVersion.Previous, Endpoint = TablesEndpoints.Tables)]
[ServiceConfiguration(ServiceVersion.Latest,   Endpoint = TablesEndpoints.Cosmos)]
[ServiceConfiguration(ServiceVersion.Previous, Endpoint = TablesEndpoints.StackTables)]
public class TablesTest { ... }
```

So we could explicitly control which combinations we want to run live, which combinations we want to record, etc.  Especially if we want to enable configurations like Docker Containers that don't support every service operation or Azure Stack which won't support the same versions at the same time.

I don't mind starting with what we have in this PR and evolving when we have concrete scenarios.

Reply:
Or maybe we create a virtual `ApplyConfiguration(test, version, param)` on the base class that we could override per service?  I could use that to enable/disable recordings (especially if we used metadata as @pakrym suggested), setup various conditions, look for other attributes like `[NotSupportedByContainer]` on certain tests and translate that into `[ExpectedException]` for that pass, split the param on commas if I wanted to jam extra fields in there, etc.  Then it's not the test framework's job to do everything for everyone.